projects/hitl/software/hitl/cancontroller.py

```diff
@@ -113,14 +119,16 @@ def _parse_frame(self, msg):
         
         Parses through a CAN frame and updates the self.states dictionary
         """
+        print(msg)
         # Get the message name
-        msg_name = self.db.get_message_by_frame_id(msg.arbitration_id)
+        msg_name = self.db.get_message_by_frame_id(msg.arbitration_id).name
         # Decode the data
         data = self.db.decode_message(msg.arbitration_id, msg.data)
+        print(data)
         # Update the state dictionary
         self.signals[msg_name].update(data)
 
-    def __del__(self):
```

Comment:
Can we keep the __del__ name? Not sure why it doesn't work yet, but fixing it doesn't need to be in the scope of the PR. If you need to delete a CANController, just manually call can.__del__() for now.